ipatests/test_webui/test_pwpolicy.py

```diff
@@ -54,4 +51,214 @@ def test_crud(self):
+                        delete=True)
+
+        # empty group and priority (requires the field)
+        self.navigate_to_entity(pwpolicy.ENTITY)
+        self.facet_button_click('add')
+        self.dialog_button_click('add')
+        self.assert_field_validation_required(field='cn')
+        self.assert_field_validation_required(field='cospriority')
+        self.close_all_dialogs()
+
+        # test delete default policy and
+        # confirming by keyboard to test ticket-4097
+        self.navigate_to_entity(pwpolicy.ENTITY)
+        self.select_record(pwpolicy.DEFAULT_POLICY)
+        self.facet_button_click('remove')
+        self.dialog_button_click('ok')
+        self.assert_last_error_dialog(EXPECTED_ERR, details=True)
+        actions = ActionChains(self.driver)
+        actions.send_keys(Keys.TAB)
+        actions.send_keys(Keys.ENTER).perform()
+        self.wait(0.5)
+        self.assert_record(pwpolicy.DEFAULT_POLICY)
+
+        # test add/delete Passwordpolicy confirming using ENTER key ticket-3200
+        self.add_record(pwpolicy.ENTITY, pwpolicy.DATA3, dialog_btn=None)
+        actions = ActionChains(self.driver)
+        actions.send_keys(Keys.ENTER).perform()
+        self.wait_for_request(d=0.5)
+        self.assert_notification(assert_text=EXPECTED_MSG)
+        self.assert_record(pwpolicy.PKEY3)
+        self.close_notifications()
+        self.delete_record(pwpolicy.PKEY3, confirm_btn=None)
+        actions = ActionChains(self.driver)
+        actions.send_keys(Keys.ENTER).perform()
+        self.wait_for_request(d=0.5)
+        self.assert_notification(assert_text='1 item(s) deleted')
+        self.assert_record(pwpolicy.PKEY3, negative=True)
+        self.close_notifications()
+
+        # test delete multiple records
+        records = [pwpolicy.DATA, pwpolicy.DATA1, pwpolicy.DATA2]
+        self.navigate_to_entity(pwpolicy.ENTITY)
+        self.select_multiple_records(records)
+        self.facet_button_click('remove')
+        self.dialog_button_click('ok')
+
+    @screenshot
+    def test_negative_value(self):
+        """
+        Negative test for Password policy fields in edit page
+        """
+        self.init_app()
+        self.navigate_to_entity(pwpolicy.ENTITY)
+        non_interger_expected_error = 'Must be an integer'
+        minimum_value_expected_error = 'Minimum value is 0'
+        non_integer = 'nonInteger'
+        maximum_value = '2147483649'
+        minimum_value = '-1'
+
+        self.add_record(pwpolicy.ENTITY, pwpolicy.DATA1,
+                        dialog_btn='add_and_edit')
+
+        for field in FIELDS:
+            # bigger than max value
+            if field == 'krbmaxpwdlife':
+                self.modify_policy(field, 'Maximum value is 20000',
+                                   maximum_value)
+            elif field == 'krbpwdmindiffchars':
+                self.modify_policy(field, 'Maximum value is 5', maximum_value)
+            else:
```

Comment:
This if, elif, else construction could be a bit misleading. Based on it, one would imagine that there are three fields but there is much more. Not sure if it can be enhanced just by comment what is the test or by changing of a structure or using a map.

Reply:
The change doesn't reflect what I meant. The added comment is just repeating the code. That part was clear. 

I meant something like:
```python
# 'cn'("group") field is read-only and all other fields are integers without special validator
else:
                self.check_expected_error(field, 'Maximum value is 2147483647',
                                          maximum_value)
```